web/src/routes/+layout.svelte

```diff
@@ -104,6 +107,32 @@
   };
 
   $effect(() => void handleRelease($release));
+
+  serverRestarting.subscribe((isRestarting) => {
+    if (!isRestarting) {
+      return;
+    }
+
+    if (isRestarting.isMaintenanceMode !== page.url.pathname.startsWith(AppRoute.MAINTENANCE)) {
+      modalManager.show(ServerRestartingModal, {}).catch((error) => console.error('Error [ServerRestartBox]:', error));
+
+      // we will be disconnected momentarily
+      // wait for reconnect then reload
+      let waiting = false;
+      websocketStore.connected.subscribe((connected) => {
+        if (!connected) {
+          waiting = true;
+        } else if (connected && waiting) {
+          void goto(
+            isRestarting.isMaintenanceMode
+              ? maintenanceCreateUrl(new URL(location.href))
+              : maintenanceReturnUrl(new URLSearchParams(location.search), location.origin),
+            { invalidateAll: true },
+          );
```

Comment:
Maybe we should just try to refresh the page, which should handle redirecting to the maintenance page